spec/licensee/projects/github_project_spec.rb

```diff
@@ -0,0 +1,140 @@
+RSpec.describe Licensee::Projects::GitHubProject do
+  describe '#initialize' do
+    subject(:instance) { described_class.new(github_url) }
+
+    context 'with a GitHub URI' do
+      let(:github_url) { 'https://github.com/benbalter/licensee' }
+
+      it 'should set @repo' do
+        expect(instance.repo).to eq('benbalter/licensee')
+      end
+    end
+
+    context 'with a GitHub git URI' do
+      let(:github_url) { 'https://github.com/benbalter/licensee.git' }
+
+      it 'should set @repo, stripping the trailing extension' do
+        expect(instance.repo).to eq('benbalter/licensee')
+      end
+    end
+
+    context 'with a non-GitHub URI' do
+      let(:github_url) { 'https://gitlab.com/benbalter/licensee' }
+
+      it 'should raise an ArgumentError' do
+        expect { instance }.to raise_error(ArgumentError)
+      end
+    end
+
+    context 'with a local folder' do
+      let(:github_url) { fixture_path('mit') }
+
+      it 'should raise an ArgumentError' do
+        expect { instance }.to raise_error(ArgumentError)
+      end
+    end
+  end
+
+  subject { described_class.new(github_url) }
+
+  let(:repo) { 'benbalter/licensee' }
+  let(:github_url) { 'https://github.com/benbalter/licensee' }
```

Comment:
I don't know if/why we'd change this, but to make the test a bit less susceptible to external change (e.g., if we move the project to `licensee/licensee`, I'd suggest making the references to `benbalter/licensee` point to a single `#{repo}` (from the let statement), wherever possible (e.g., it won't be possible in the before block).